lib/stack_master/template_compiler.rb

```diff
@@ -1,8 +1,11 @@
 module StackMaster
   class TemplateCompiler
+    TemplateCompilationFailed = Class.new(RuntimeError)
 
     def self.compile(config, template_file_path)
       template_compiler_for_file(template_file_path, config).compile(template_file_path)
+    rescue StandardError
```

Comment:
The intention is to be more generic here, each new template compiler will have it's own errors, and if they failed (in one way or another), we wrap in our SM compile error and can handle it more generically upstream (in cli.rb)...

@thekindofme we should make sure the errors `.cause` message is being outputted to STDERR. Otherwise it won't be obvious why compilation failed. Cause is the wrapped error, automatically provided by ruby.